### ISSUE TITLE
Notification when someone follows you

### ISSUE THREAD

**Issue by local-minimum** [ludumdare/ludumdare#1489]
It feels like a bit of asymmetry at the moment.

If I follow someone, they will not know I follow them.
If that someone follows me, they will know I had already followed them, yet I will not know we are friends (unless I specifically look them up in friends list).

**Comment by sgstair** [comment]
Not a bad idea; it's pretty easy to implement.

**Comment by local-minimum** [comment]
This would come with a new state on the notification then I hope so it is easy to separate out from the rest? It will require some handling of the UI because I'm not sure what will happen if you get an unexpected notification type.

**Comment by sgstair** [comment]
yeah, not that the existing code pays any attention to the state currently - but yes, it'd be a new notification type.

**Comment by povrazor** [comment]
Speaking of which, there should be a default icon for an unknown type (if there isn't already).

**Comment by local-minimum** [comment]
What would the notification say?  "You also have 2 notifications unknown to the UI relating to xxx classified as yyy"

**Comment by povrazor** [comment]
What would happen now if a notification with an unknown type was added? 

I basically just mean have a generic notification for types we don't know yet. That way if we decide to add some new notifications without a UI, we still see them, instead of sitting there wondering "why don't I see my notification"?

**Comment by sgstair** [comment]
At the moment the notification code doesn't particularly care at all about notification type, it infers the type from the fields (note/node) and whether there is a mention in the content. So e.g. a notification for "x subscribed to you" would probably appear as "a new user was posted"

**Comment by povrazor** [comment]
Sorry, I inferred from the discussions that there was a `type` field in notification that suggested the origin of the notification, and that was being used to decide how to display it.

**Comment by sgstair** [comment]
there is a type field, it is not being used at the moment :) This will probably need to change.

**Comment by local-minimum** [comment]
Yes it could clean up the code a lot. I was just eager to catch mentions and friends' activities from start.
